autograder/handgrading/views/handgrading_result_views.py

```diff
@@ -85,7 +89,34 @@ class HandgradingResultView(NestedModelView):
     @handle_object_does_not_exist_404
     def get(self, request, *args, **kwargs):
         group: ag_models.Group = self.get_object()
-        return response.Response(group.handgrading_result.to_dict())
+        result: dict = group.handgrading_result.to_dict()
+
+        # Only filter the output if the user doesn't have privilege
+        # Can be changed to is_student, if the permission does not inherit (i.e. if admin/staff are not considered student)
+        can_access_all_rubrics = (
+            P(HandgradingAllRubricReleased)
+            | P(is_admin)
+            | P(is_staff)
+            | P(is_handgrader)
+        )
```

Comment:
I'd prefer not to use the permissions objects in this way, can we change this to something like this?
```
course = group.project.course
show_only_applied_rubric = (
    group.project.handgrading_rubric.show_only_applied_rubric_to_students
    and not course.is_staff(request.user)  # Admins count as staff
    and not course.is_handgrader(request.user)
)
```

Reply:
Resolved in a7a93614449817ad8cc0da1aafcad2a34ce38832